junit/src/main/java/net/neoforged/gradle/junit/JUnitProjectPlugin.java

```diff
@@ -0,0 +1,157 @@
+package net.neoforged.gradle.junit;
+
+import net.neoforged.gradle.common.runs.run.RunImpl;
+import net.neoforged.gradle.common.runtime.definition.CommonRuntimeDefinition;
+import net.neoforged.gradle.common.util.SourceSetUtils;
+import net.neoforged.gradle.common.util.TaskDependencyUtils;
+import net.neoforged.gradle.common.util.exceptions.MultipleDefinitionsFoundException;
+import net.neoforged.gradle.common.util.exceptions.NoDefinitionsFoundException;
+import net.neoforged.gradle.common.util.run.RunsUtil;
+import net.neoforged.gradle.dsl.common.extensions.Minecraft;
+import net.neoforged.gradle.dsl.common.runs.task.extensions.TestTaskExtension;
+import net.neoforged.gradle.userdev.runtime.definition.UserDevRuntimeDefinition;
+import org.gradle.api.GradleException;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.dsl.DependencyFactory;
+import org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestFramework;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.TaskProvider;
+import org.gradle.api.tasks.testing.Test;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class JUnitProjectPlugin implements Plugin<Project> {
+    private static final Logger LOG = LoggerFactory.getLogger(JUnitProjectPlugin.class);
+
+    @Override
+    public void apply(Project project) {
+        project.getTasks().named("test").configure(task -> task.getExtensions().create(TestTaskExtension.NAME, TestTaskExtension.class, project));
+        project.afterEvaluate(this::applyAfterEvaluate);
+    }
+
+    private void applyAfterEvaluate(Project project) {
+        Test testTask = (Test) project.getTasks().getByName("test");
```

Comment:
This should probably be ran for all test tasks.
Or some kind of extension should exist within the project to control what test tasks it should target.